tgg/annotations/get_ensembl_db_info.py

```diff
@@ -22,53 +24,168 @@
+                "gene.biotype",
+                "gene.created_date",
+                "gene.modified_date",
+
+                "transcript.stable_id",
+                "transcript.biotype",
+                "transcript.created_date",
+                "transcript.modified_date",
+            ]
+
             columns_str = ", ".join(columns)
-            c.execute(f"SELECT {columns_str} FROM gene")
+            query_string = f"SELECT {columns_str} FROM gene LEFT JOIN transcript ON {join_clause}"
+            if only_protein_coding:
+                query_string += " WHERE gene.biotype = 'protein_coding' AND transcript.biotype = 'protein_coding'"
 
-            for row in c:
-                d = dict(zip(columns, row))
-                gene_id = d['stable_id']
-                gene_id_to_dates[gene_id] = d['created_date'], d['modified_date']
+            cursor.execute(query_string)
 
-    return gene_id_to_dates
+            for row in cursor:
+                gene_and_transcript_info = dict(zip(columns, row))
+                gene_id = gene_and_transcript_info['gene.stable_id']
+                gene_id_to_transcript_id[gene_id].append(gene_and_transcript_info)
 
+    return gene_id_to_transcript_id
 
-def get_canonical_transcript_created_modified_dates(database=CURRENT_ENSEMBL_DATABASE):
-    transcript_id_to_dates = {}
-    with pymysql.connect(host="useastdb.ensembl.org", user="anonymous", database=database) as conn:
-        with conn.cursor() as c:
-            columns = ["transcript.stable_id", "transcript.created_date", "transcript.modified_date"]
-            columns_str = ", ".join(columns)
-            c.execute(f"SELECT {columns_str} FROM gene LEFT JOIN transcript ON canonical_transcript_id = transcript_id")
 
-            for row in c:
-                d = dict(zip(columns, row))
-                gene_id = d['transcript.stable_id']
-                transcript_id_to_dates[gene_id] = d['transcript.created_date'], d['transcript.modified_date']
+def get_gene_id_to_transcript_ids(
+        database=CURRENT_ENSEMBL_DATABASE,
+        only_protein_coding=False,
+        only_canonical_transcripts=False):
+    """Returns a dictionary mapping each Ensembl gene_id => a list of transcript ids for that gene.
+
+    Args:
+        database (str): The Ensembl database name (eg. "homo_sapiens_core_107_38")
+        only_protein_coding (bool): If True, only return protein coding genes
+        only_canonical_transcripts (bool): If True, only return canonical transcripts
+    Return:
+        dict: mapping ENSG id string to a list of ENST id strings
+    """
+
+    gene_id_to_transcript_metadata_list = get_gene_id_to_transcript_metadata(
+        database=database,
+        only_canonical_transcripts=only_canonical_transcripts,
+        only_protein_coding=only_protein_coding)
+
+    return {
+        gene_id: [
+            transcript_metadata["transcript.stable_id"] for transcript_metadata in transcript_metadata_list
```

Comment:
is this name stable: `transcript.stable_id`? should this be an argument, something like:
```
transcript_field: str = 'transcript.stable_id'
```

Reply:
I prefer to make these internal fields configurable when specific use-cases come up. This keeps the interface simple until the added complexity is needed.